src/main/java/appeng/core/sync/packets/PacketNEIRecipe.java

```diff
@@ -62,117 +61,297 @@
+	// currentItem = craftMatrix.getStackInSlot( x );
+	// }
+	// }
+	// }
+	//
+	// // True if we need to fetch an item for the recipe
+	// if( patternItem != null && currentItem == null )
+	// {
+	// // Grab from network by recipe
+	// ItemStack whichItem = Platform.extractItemsByRecipe( energy, cct.getSource(), storage, player.worldObj, r, is,
+	// testInv, patternItem, x, all, realForFake, filter );
+	//
+	// // If that doesn't get it, grab exact items from network (?)
+	// // TODO see if this code is necessary
+	// if( whichItem == null )
+	// {
+	// for( int y = 0; y < this.recipeStack[x].length; y++ )
+	// {
+	// IAEItemStack request = AEItemStack.create( this.recipeStack[x][y] );
+	// if( request != null )
+	// {
+	// if( filter == null || filter.isListed( request ) )
+	// {
+	// request.setStackSize( 1 );
+	// IAEItemStack out = Platform.poweredExtraction( energy, storage, request, cct.getSource() );
+	// if( out != null )
+	// {
+	// whichItem = out.getItemStack();
+	// break;
+	// }
+	// }
+	// }
+	// }
+	// }
+	//
+	// // If that doesn't work, grab from the player's inventory
+	// if( whichItem == null && playerInventory != null )
+	// {
+	// whichItem = this.extractItemFromPlayerInventory( player, realForFake, patternItem );
+	// }
+	//
+	// craftMatrix.setInventorySlotContents( x, whichItem );
+	// }
+	// }
+	// con.onCraftMatrixChanged( craftMatrix );
+	// }
+	// }
+	// }
+	// }
+	// }
+	// }
+
+	/**
+	 * Tries to extract an item from the player inventory. Does account for fuzzy items.
+	 *
+	 * @param player the {@link EntityPlayer} to extract from
+	 * @param mode the {@link Actionable} to simulate or modulate the operation
+	 * @param patternItem which {@link ItemStack} to extract
+	 * @return null or a found {@link ItemStack}
+	 */
+	private ItemStack extractItemFromPlayerInventory( EntityPlayer player, Actionable mode, ItemStack patternItem )
+	{
+		final InventoryAdaptor ia = InventoryAdaptor.getAdaptor( player, ForgeDirection.UNKNOWN );
+		final AEItemStack request = AEItemStack.create( patternItem );
+		final boolean isSimulated = mode == Actionable.SIMULATE;
+		final boolean checkFuzzy = request.isOre() || patternItem.getItemDamage() == OreDictionary.WILDCARD_VALUE || patternItem.hasTagCompound() || patternItem.isItemStackDamageable();
+
+		if( !checkFuzzy )
 		{
-			this.recipe = new ItemStack[9][];
-			for( int x = 0; x < this.recipe.length; x++ )
+			if( isSimulated )
+			{
+				return ia.simulateRemove( 1, patternItem, null );
+			}
+			else
+			{
+				return ia.removeItems( 1, patternItem, null );
+			}
+		}
+		else
+		{
+			if( isSimulated )
 			{
-				NBTTagList list = comp.getTagList( "#" + x, 10 );
+				return ia.simulateSimilarRemove( 1, patternItem, FuzzyMode.IGNORE_ALL, null );
+			}
+			else
+			{
+				return ia.removeSimilarItems( 1, patternItem, FuzzyMode.IGNORE_ALL, null );
+			}
+		}
+	}
+
+	@Override
+	public void fromBytes( ByteBuf buf )
+	{
+		final NBTTagCompound recipe = ByteBufUtils.readTag( buf );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "recipe" which hides the field declared at line 68. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)